crates/ruma-client-api/src/dehydrated_device/put_dehydrated_device.rs

```diff
@@ -0,0 +1,90 @@
+//! `PUT /_matrix/client/*/dehydrated_device/`
+//!
+//! Uploads a dehydrated device to the homeserver.
+
+pub mod unstable {
+    //! `msc3814` ([MSC])
+    //!
+    //! [MSC]: https://github.com/uhoreg/matrix-doc/blob/shrivelled_sessions/proposals/3814-dehydrated-devices-with-ssss.md
+
+    use std::collections::BTreeMap;
+
+    use ruma_common::{
+        api::{request, response, Metadata},
+        encryption::{DeviceKeys, OneTimeKey},
+        metadata,
+        serde::Raw,
+        OwnedDeviceId, OwnedDeviceKeyId,
+    };
+
+    use crate::dehydrated_device::DehydratedDeviceData;
+
+    const METADATA: Metadata = metadata! {
+        method: PUT,
+        rate_limited: false,
+        authentication: AccessToken,
+        history: {
+            unstable => "/_matrix/client/unstable/org.matrix.msc3814.v1/dehydrated_device",
+        }
+    };
+
+    /// Request type for the `PUT` `dehydrated_device` endpoint.
+    #[request(error = crate::Error)]
+    pub struct Request {
+        /// The unique ID of the device.
+        pub device_id: OwnedDeviceId,
+
+        /// The display name of the device.
+        pub initial_device_display_name: String,
```

Comment:
This is optional in the MSC

Reply:
Fixed in e83d6e9e2c3cdd1f9f7a16a8f5895a3de41c5454.